src/main/org/codehaus/groovy/classgen/AsmClassGenerator.java

```diff
@@ -1218,7 +1218,7 @@ public void visitVariableExpression(VariableExpression expression) {
 
         BytecodeVariable variable = controller.getCompileStack().getVariable(variableName, false);
         if (variable == null) {
-            processClassVariable(variableName);
+            processClassVariable(expression);
```

Comment:
this PR is closed, as I have mentioned before i am planning to do some refactoring and creating a new PR, I will make expression final there